### DISCUSSION TITLE
Recording keyboard input while response choice is set to NO_Keys

### DISCUSSION THREAD

**Discussion by samuelcpsych** [jspsych/jsPsych#1990] [Q&A]
Hi,
I'm trying to record the pressing of the space key in image-keyboard-response trials. The reason for not just enabling it in the plugin (and defining the choices with e.g. ALL_KEYS), is that I want to show every stimulus only for a specific number of seconds without any influence on the exposure duration by the participants. Recording the key input is crucial because in one condition I want to increase the attention toward the duration of stimuli exposure - so the recording of keyboard responses should just help to insure that participants followed the instructions.

Since I could not make it work inside the plugin, my current, very simple workaround is:
```
    value = 0;
    space_bar = 32;
    x = window.onkeyup = function(e){
        if(e.keyCode === space_bar){
            value++
        };
    };
```
The problem with this solution is that I can only count the number of "key-up" events without having any information about               the key-press duration or about the trial in which the key was being pressed.

I am looking forward for any help or ideas on this problem! Thank you very much in advance!

**Comment by jodeleeuw** [comment]
Hi @samuelcpsych,

What about using the `response_ends_trial` parameter in `image-keyboard-response`? Does that solve your use case?

```js
var trial = {
	type: 'image-keyboard-response',
	stimulus: 'image.png',
	trial_duration: 4000,
	response_ends_trial: false,
	choices: [' ']
}
```

**Reply by samuelcpsych** [reply]
Hi @jodeleeuw ,

Thank you very much for your answer, this helps me a lot. I remember trying out this parameter before, but something was not working in my code correctly at that time, now it works perfectly.

Is there any way to additionally save the duration of the key pressing in the ```image-keyboard-response``` plugin?
